src/graphcut/normalized_cut.jl

```diff
@@ -115,7 +115,7 @@ function _partition_weightmx(cut, W::SparseMatrixCSC)
     return (W1, W2, vmap1, vmap2)
 end
 
-function _recursive_normalized_cut(W, thres=thres, num_cuts=num_cuts)
+function _recursive_normalized_cut(W, thres, num_cuts)
     m, n = size(W)
     D = Diagonal(vec(sum(W; dims=2)))
 
```

Comment:
I don't understand how this even compiled before

Reply:
Not sure either. That's why JET is a good thing to use by default, even if it only catches typos like that

Reply:
Ah, I think Julia creates multiple methods - one where one has to specify all positional args, and that one also runs, but if one tries to omit a positional argument, then it fails. Still weird though...
```
julia> function f(a, b=b)
       return a + b
       end
f (generic function with 2 methods)

julia> f(1, 2)
3

julia> f(1)
ERROR: UndefVarError: `b` not defined
Stacktrace:
 [1] f(a::Int64)
   @ Main ./REPL[4]:2
 [2] top-level scope
   @ REPL[7]:1

julia> methods(f)
# 2 methods for generic function "f" from Main:
 [1] f(a)
     @ REPL[4]:1
 [2] f(a, b)
     @ REPL[4]:1
```

Reply:
Yes but the thing that JET caught is that the equivalent of `b` was not defined anywhere in the Graphs code